### PR TITLE
[WIP] Make lerna version understand --scope argument

### PR DESCRIPTION
**Pull request by dentuzhik**
*Sorry for not folllowing a template to the fullest yet, this is still a WIP, and I would like some feedback first before moving on*

## Description
This PR implements `--scope` argument for `lerna version` (and seems like subsequently `lerna publish` too).

## Motivation and Context
https://github.com/lerna/lerna/issues/2512

## How Has This Been Tested?
All existing test for `commands/version` are passing.
Test for `command/publish` which checks `--scope` not available, obviously fails.

I tested locally via CLI linking, it works as expected for my use case, although I am not 100% yet certain whether it will work as expected for cases when a version has to be updated in packages outside `--scope` argument.

Quickly tried to unit-test, but run into a problem that `collectUpdates` function is completely mocked, and to test it properly I need to unmock the function, which might be a potentially breaking (and quite large) change to current test suite, so I decided not to follow that road yet.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: commands/version/__tests__/version-command.test.js [hunk 1]
```diff
@@ -232,6 +232,20 @@ describe("VersionCommand", () => {
     });
   });
 
+  describe("--scope", () => {
+    // TODO: This test makes no sense because of collect-updates mock
+    it("should change versiosn only for scoped packages", async () => {
```
Comment (dentuzhik) [inline commands/version/__tests__/version-command.test.js:237]: Just for some inline context, this mock for `collect-updates` does not currently allow to write any meaningful tests:
https://github.com/lerna/lerna/blob/master/commands/__mocks__/@lerna/collect-updates.js#L9
image

**Comment by dentuzhik** [comment]
@evocateur 

I would appreciate if you could help me out a little bit and give some guidance, in terms of whether guys you even want this option available for this command, and whether unmocking 

I see potential concerns over how scoping is going to affect dependencies, but rather have some feedback from you on this. E.g package matches `--scope` in terms of updating version, but acts a dependency for another package which is outside of `--scope` matching.

**Comment by simllll** [comment]
Do you have any prepublished version that I can use in the meantime? I'm having a simiilar issue, and would like to test if it solves my issues as well :+1: 
Thanks for your work!

**Comment by dentuzhik** [comment]
@simllll 

I haven't touched this in a while, since there's no feedback, but if you are interested I can try to re-publish lerna as a scoped package and you can give it a go. Alternatively you can also clone the branch, and re-publish under your own scope in npm :)

**Pull request closed without merging**